Sources/TuistCore/MetadataProviders/CodesignController.swift

```diff
@@ -0,0 +1,58 @@
+import Command
+import Foundation
+import Mockable
+import Path
+
+@Mockable
+/// Utility to interact with the `codesign` CLI.
+public protocol CodesignControlling {
+    /// Provides the signature of the XCFramework at the given `xcframeworkPath`, or `nil` if unsigned.
+    func codesignSignature(of xcframeworkPath: AbsolutePath) async throws -> String?
```

Comment:
nit: the `codesign` prefix of methods here feels slightly redundant since we're operating in a `CodesignController`